src/ngraph/runtime/cpu/builder/reduction.hpp

```diff
@@ -41,10 +41,10 @@
         return;                                                                                    \
     }                                                                                              \
                                                                                                    \
-    if (reduction_axes.size() == arg_rank)                                                         \
+    if (reduction_axes.size() == arg_rank && is_fp_i64(args[0].get_element_type()))                \
```

Comment:
You could make an overload of `is_fp_i64` which takes an arg instead of an element type. Might make the code more readable.